### PR TITLE
jq_fadeIn_or_out_list_in_circle

### PR DESCRIPTION
**Pull request by lord-sumit**

### CODE REVIEW

File: jquery/exercise8/js/slideshow.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+  };
+};
+
+SlideShow.prototype.insertElement = function(insertIn, moveElement) {
+  insertIn.prepend(moveElement);
+};
+
+$(document).ready(function(){
+  var slideshow = new SlideShow($("body"), $("#slideshow"), $("#myList"));
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: Why you need to select and pass body?

File: jquery/exercise8/js/slideshow.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+function SlideShow(body, slideShow, list) {
+  this.$body = body;
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: Body of document is always constant.
you don't need to store it.

File: jquery/exercise8/js/slideshow.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  this.$slideShowElement = slideShow;
+  this.$list = list.children();
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: If it storing children of list then why it's name is list?

**Review by rajatbansal93 — CHANGES_REQUESTED** [review]

File: jquery/exercise8/js/slideshow.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      $(this).css("display", "list-item");
+    }
+  };
+};
+
+SlideShow.prototype.insertElement = function(insertIn, moveElement) {
+  insertIn.prepend(moveElement);
+};
+
+$(document).ready(function(){
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: spacing issue

File: jquery/exercise8/js/slideshow.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  this.insertElement(this.$body, this.$slideShowElement);
+  /*2.Write code to cycle through the list items inside the element;
+      fade one in, display it for a few seconds, then fade it out and fade in the next one
+    3.When you get to the end of the list, start again at the beginning.*/
+  this.$list.first().fadeToggle(300,this.fadeInOutElement());
+  /*4.For an extra challenge, create a navigation area under the slideshow
+      that shows how many images there are and which image you're currently viewing.*/
+};
+
+SlideShow.prototype.fadeInOutElement = function(first_argument) {
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: first_argument?

and in second cycle also it's first_argument?

File: jquery/exercise8/js/slideshow.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  var _this = this;
+  return function () {
+    if($(this).text() == _this.$list.last().text()){
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: lot of redundacny

**Review by rajatbansal93 — CHANGES_REQUESTED** [review]

File: jquery/exercise8/js/slideshow.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      _this.$list.first().fadeToggle(300,_this.fadeInOutElement());
+    }
+    else{
+      $(this).next().fadeToggle(300,_this.fadeInOutElement());
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: spacing issues

File: jquery/exercise8/js/slideshow.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  this.$list.first().fadeToggle(300,this.fadeInOutElement());
+  /*4.For an extra challenge, create a navigation area under the slideshow
+      that shows how many images there are and which image you're currently viewing.*/
+};
+
+SlideShow.prototype.fadeInOutElement = function(first_argument) {
+  var _this = this;
+  return function () {
+    if($(this).text() == _this.$list.last().text()){
+      _this.$list.first().fadeToggle(300,_this.fadeInOutElement());
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: spacing issue

File: jquery/exercise8/js/slideshow.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    else{
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: spacing issue

**Review by rajatbansal93 — CHANGES_REQUESTED** [review]

File: jquery/exercise8/js/slideshow.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  this.$list = list.children();
+};
+
+SlideShow.prototype.bindEvents = function() {
+  /*1.Move the #slideshow element to the top of the body.*/
+  this.insertElement(this.$body, this.$slideShowElement);
+  /*2.Write code to cycle through the list items inside the element;
+      fade one in, display it for a few seconds, then fade it out and fade in the next one
+    3.When you get to the end of the list, start again at the beginning.*/
+  this.$list.first().fadeToggle(300,this.fadeInOutElement());
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: 300 should go in a constant in prototype

File: jquery/exercise8/js/slideshow.js [hunk 11]
```diff
@@ -0,0 +1,41 @@
+function SlideShow(slideShow, list) {
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: why not starting with $

File: jquery/exercise8/js/slideshow.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+  this.$slideShowElement = slideShow;
+  this.$childrenOfList = list.children();
+};
+
+SlideShow.prototype.during = 300;
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: Constant start with capital letter, it should be During

File: jquery/exercise8/js/slideshow.js [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+
+SlideShow.prototype.bindEvents = function() {
+  /*1.Move the #slideshow element to the top of the body.*/
+  this.insertElement($("body"), this.$slideShowElement);
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: why are you doing things other than binding events in this function?

File: jquery/exercise8/js/slideshow.js [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+  /*2.Write code to cycle through the list items inside the element;
+      fade one in, display it for a few seconds, then fade it out and fade in the next one
+    3.When you get to the end of the list, start again at the beginning.*/
+  this.$childrenOfList.first().fadeToggle(this.during, this.fadeInOutElement());
+  /*4.For an extra challenge, create a navigation area under the slideshow
+      that shows how many images there are and which image you're currently viewing.*/
+};
+
+SlideShow.prototype.fadeInOutElement = function() {
```
Comment (rajatbansal93) [inline jquery/exercise8/js/slideshow.js]: take out seperate methods for fad in and out

**Review by rajatbansal93 — CHANGES_REQUESTED** [review]

File: jquery/exercise8/js/slideshow.js (hunk 14 above)
Comment (lord-sumit) [reply jquery/exercise8/js/slideshow.js]: brother i use toggle method for fade in and out due to which i cann't break it

**Pull request closed without merging**